models/magiceden/solana/magiceden_solana_trades.sql

```diff
@@ -0,0 +1,231 @@
+                , cast(json_value(args, 'strict $.OCPExecuteSaleV2Args.takerFeeBp') as double) as takerFeeBp
+                , cast(bytearray_to_bigint(bytearray_reverse(bytearray_substring(call_data,1+18,2))) as double) as takerFeeRaw
+                , call_outer_instruction_index
+                , call_inner_instruction_index
+                , call_block_time
+                , call_block_slot
+                , call_tx_id
+                , call_tx_signer
+                , row_number() over (partition by call_tx_id order by call_outer_instruction_index asc, call_inner_instruction_index asc) as call_order
+            FROM {{ source('magic_eden_solana','m2_call_ocpExecuteSaleV2') }}
+        ) trade
+        --this shortcut ONLY works if you know that a log is only emitted ONCE per call.
+        LEFT JOIN royalty_logs rl ON trade.call_tx_id = rl.call_tx_id
+            AND trade.call_block_slot = rl.call_block_slot
+            AND trade.call_order = rl.log_order 
+        {% if is_incremental() %}
+        WHERE trade.call_block_time >= NOW() - interval '7' day
+        {% endif %}
+        -- --QA
+        -- WHERE trade.call_tx_id = '4RrNYCNkJCLjvj4unGgo6owJeEg5qxWZJLYnfyiJgFVypashPaJFVgQRivEvcxeEt3xrQUiUM9bm7VP76XtPenxw'
+        -- AND trade.call_block_slot = 219668172
+    )
+    
+    , raw_nft_trades as (
+        SELECT
+            'solana' as blockchain
+            , 'magiceden' as project
+            , 'v2' as version
+            , date_trunc('day', t.call_block_time) as block_date
+            , date_trunc('month', t.call_block_time) as block_month
+            , t.call_block_time as block_time
+            , tk.token_name
+            , tk.token_symbol
+            , 'secondary' as trade_type
+            , token_size as number_of_items --all single trades right now
+            , t.trade_category
+            , t.account_buyer as buyer
+            , t.account_seller as seller
+            , t.price as amount_raw
+            , t.price/1e9 as amount_original
+            , t.price/1e9 * sol_p.price as amount_usd
+            , 'SOL' as currency_symbol
+            , 'So11111111111111111111111111111111111111112' as currency_address
+            , t.account_metadata --token id equivalent
+            , tk.account_master_edition --token id equivalent
+            , tk.account_mint --token id equivalent
+            , tk.verified_creator --contract address equivalent
+            , tk.collection_mint --contract address equivalent
+            , 'TSWAPaqyCSx2KABk68Shruf4rp7CxcNi8hAsbdwmHbN' as project_program_id
+            , cast(null as varchar) as aggregator_name
+            , cast(null as varchar) as aggregator_address
+            , t.call_tx_id as tx_id
+            , t.call_block_slot as block_slot
+            , t.call_tx_signer as tx_signer
+            , t.taker_fee as taker_fee_amount_raw --taker fees = platform fees
```

Comment:
if `taker fees = platform fees`, is there a reason to not adopt the `platform_fee_(..)` column names from nft.trades?

Reply:
Technically positive maker fees count towards platform fees. It is a weird nuance, i can call it out more in the column descriptions